core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/WindowsShareCopy.java

```diff
@@ -0,0 +1,289 @@
+   */
+  public class WindowsShareCopyConfig extends PluginConfig {
+
+    @Description("Specifies the NetBios domain name.")
+    @Nullable
+    @Macro
+    private final String netBiosDomainName;
+
+    @Description("Specifies the NetBios hostname to import files from")
+    @Macro
+    private final String netBiosHostname;
+
+    @Description("Specifies the NetBios username to user for importing files from share")
+    @Macro
+    private final String netBiosUsername;
+
+    @Description("Specifies the NetBios password for importing files from windows share")
+    @Macro
+    private final String netBiosPassword;
+
+    @Description("Specifies the NetBios share name")
+    @Macro
+    private final String netBiosSharename;
+
+    @Description("Specifies the number of threads to execute the copy operation")
+    @Nullable
+    private final Integer numThreads;
+
+    @Description("Specifies if the files needs to be overwritten or not")
+    @Nullable
+    private final Boolean overwrite;
+
+    @Description("Specifies the NetBios directory")
+    @Macro
+    private final String sourceDirectory;
+
+    @Description("The valid, full HDFS destination path in the same cluster where the file or files are to be moved. " +
+      "If a directory is specified with a file sourcePath, the file will be put into that directory. If sourcePath " +
+      "is a directory, it is assumed that destPath is also a directory. HDFSAction will not catch this inconsistency.")
+    @Macro
+    private final String destinationDirectory;
+
+    @Description("The i/o buffer size used in copying bufferstream and to write to the hdfs file. The values should" +
+      " be in the multiples of 4096. Default value is 4096")
+    @Nullable
+    private final Integer bufferSize;
+
+    @VisibleForTesting
+    WindowsShareCopyConfig(String netBiosDomainName, String netBiosHostname, String netBiosUsername,
+                           String netBiosPassword, String netBiosSharename, String sourceDirectory,
+                           String destinationDirectory, Integer bufferSize, Integer numThreads, String overwrite) {
+
+      this.netBiosDomainName = netBiosDomainName;
+      this.netBiosHostname = netBiosHostname;
+      this.netBiosUsername = netBiosUsername;
+      this.netBiosPassword = netBiosPassword;
+      this.netBiosSharename = netBiosSharename;
+      this.sourceDirectory = sourceDirectory;
+      this.destinationDirectory = destinationDirectory;
+      this.bufferSize = bufferSize;
+      this.numThreads = numThreads;
+      this.overwrite = !("false".equals(overwrite));
+    }
+
+    public Integer getNumThreads() {
+      if (numThreads == null) {
+        return MIN_NUM_THREADS;
+      }
+      return (numThreads < MIN_NUM_THREADS) ? MIN_NUM_THREADS : numThreads;
+    }
+
+    public Integer getBufferSize() {
+      if (bufferSize == null) {
+        return MIN_BUFFER_SIZE;
+      }
+      return (bufferSize < MIN_BUFFER_SIZE) ? MIN_BUFFER_SIZE : bufferSize;
```

Comment:
Why not just 
(bufferSize == null || bufferSize < MIN_BUFFER_SIZE) ? MIN_BUFFER_SIZE : bufferSize